go/parquet/internal/encoding/encoding_test.go

```diff
@@ -0,0 +1,684 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encoding_test
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"unsafe"
+
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/bitutil"
+	"github.com/apache/arrow/go/arrow/memory"
+	"github.com/apache/arrow/go/parquet"
+	"github.com/apache/arrow/go/parquet/internal/encoding"
+	"github.com/apache/arrow/go/parquet/internal/testutils"
+	"github.com/apache/arrow/go/parquet/schema"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type nodeFactory func(string, parquet.Repetition, int32) *schema.PrimitiveNode
+
+func createNodeFactory(t reflect.Type) nodeFactory {
+	switch t {
+	case reflect.TypeOf(true):
+		return schema.NewBooleanNode
+	case reflect.TypeOf(int32(0)):
+		return schema.NewInt32Node
+	case reflect.TypeOf(int64(0)):
+		return schema.NewInt64Node
+	case reflect.TypeOf(parquet.Int96{}):
+		return schema.NewInt96Node
+	case reflect.TypeOf(float32(0)):
+		return schema.NewFloat32Node
+	case reflect.TypeOf(float64(0)):
+		return schema.NewFloat64Node
+	case reflect.TypeOf(parquet.ByteArray{}):
+		return schema.NewByteArrayNode
+	case reflect.TypeOf(parquet.FixedLenByteArray{}):
+		return func(name string, rep parquet.Repetition, field int32) *schema.PrimitiveNode {
+			return schema.NewFixedLenByteArrayNode(name, rep, 12, field)
+		}
+	}
+	return nil
```

Comment:
is it a common pattern to return nil for unhandled enum cases in go (and not have an error or compiler detect unhandled cases?)

Reply:
NM, this is testing code should be fine.